### PR TITLE
support nvm installed by homebrew on Mac

### PR DESCRIPTION
**Pull request by 3cp**

### CODE REVIEW

File: init.fish [hunk 1]
```diff
@@ -3,7 +3,14 @@ function init -a path --on-event init_nvm
 
   if available fenv
     set -q NVM_DIR; or set -gx NVM_DIR ~/.nvm
-    fenv source $NVM_DIR/nvm.sh >/dev/null ^&1
+
+    if available brew; and brew --prefix nvm > /dev/null
```
Comment (derekstavis) [inline init.fish]: This doesn't work for me. Even when `nvm` isn't installed, `brew --prefix nvm` outputs the prefix. You could use `brew ls --versions nvm`

File: init.fish [hunk 2]
```diff
@@ -3,7 +3,14 @@ function init -a path --on-event init_nvm
...
+      set nvm_runnable (brew --prefix nvm)
```
Comment (derekstavis) [inline init.fish]: I would use a global variable here `set -g nvm_prefix (brew --prefix nvm)`, so that we don't need to check it again in `nvm` function.

File: init.fish [hunk 3]
```diff
@@ -3,7 +3,14 @@ function init -a path --on-event init_nvm
...
+    else
+      set nvm_runnable $NVM_DIR
```
Comment (derekstavis) [inline init.fish]: `set -g nvm_prefix $NVM_DIR`

File: init.fish [hunk 4]
```diff
@@ -3,7 +3,14 @@ function init -a path --on-event init_nvm
...
+    end
+
+    fenv source $nvm_runnable/nvm.sh >/dev/null ^&1
```
Comment (derekstavis) [inline init.fish]: `fenv source $nvm_prefix/nvm.sh >/dev/null ^&1`

File: functions/nvm.fish [hunk 5]
```diff
@@ -1,11 +1,17 @@
 function nvm -d "Node version manager"
-  if test -e $NVM_DIR/nvm.sh
+  if available brew; and brew --prefix nvm > /dev/null
+    set nvm_runnable (brew --prefix nvm)
```
Comment (derekstavis) [inline functions/nvm.fish]: See the note below about using global variables, would save us some seconds in avoiding running this check every time the command `nvm` is issued.

File: .install-linux.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
```
Comment (derekstavis) [inline .install-linux.sh]: Add a newline after the shebang.

File: .install-osx.sh [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+brew update
+brew install fish nvm
+mkdir ~/.nvm
+curl -L https://github.com/oh-my-fish/oh-my-fish/raw/master/bin/install | fish
```
Comment (derekstavis) [inline .install-osx.sh]: Use `OMF_INSTALLER` environment variable, keep it on `travis.yml`

File: .travis.yml [hunk 8]
```diff
@@ -1,35 +1,13 @@
...
 
-sudo: false
+os:
+  - linux
+  - osx
 
-env:
-  global:
-    - FISH_RELEASE="http://fishshell.com/files/2.2.0/fish-2.2.0.tar.gz"
-    - OMF_INSTALLER="https://raw.githubusercontent.com/oh-my-fish/oh-my-fish/master/bin/install"
```
Comment (derekstavis) [inline .travis.yml:8]: Keep this environment variable here. It'll be used on both install scripts.

File: .install-linux.sh [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+FISH_RELEASE="http://fishshell.com/files/2.2.0/fish-2.2.0.tar.gz"
+OMF_INSTALLER="https://raw.githubusercontent.com/oh-my-fish/oh-my-fish/master/bin/install"
```
Comment (derekstavis) [inline .install-linux.sh]: Keep this environment variable in `travis.yml`, see below.

**Comment by derekstavis** [comment]
Hey @huochunpeng, thanks a lot for the PR. Although I'm not using nvm through `brew` this is very welcome to the plugin. I've done some considerations on the PR which I would like to address before merging it.

File: .install-osx.sh [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+fish -c 'omf install foreign-env'
+mkdir -p $HOME/.local/share/omf/pkg/nvm && cp -R * $HOME/.local/share/omf/pkg/nvm
+fish -c "nvm install v4.2.4; and nvm alias default v4.2.4"
```
Comment (derekstavis) [inline .install-osx.sh:9]: Why is this needed?

File: init.fish (hunk 1 above)
Comment (3cp) [inline init.fish]: you are right, I tried the negative side in a wrong way (brew --prefix something_doesnt_in_homebrew).

File: init.fish (hunk 3 above)
Comment (3cp) [inline init.fish]: I am not sure do we need set -gx to export the variable?

Comment (derekstavis) [inline init.fish]: I think that prefix isn't needed to be exported. Just making it global (`set -g`) looks sufficient.

File: .install-osx.sh (hunk 10 above)
Comment (3cp) [inline .install-osx.sh:9]: With nvm installed by brew, if there is no default node installed, nvm doesn't set NVM_BIN.
I will try the standalone nvm installer on Mac.

Comment (derekstavis) [inline .install-osx.sh:9]: Ah, I see. I believe this is needed, as node isn't available on OS X out-of-box. My approach in this case is a little different: I always do a `brew install node` to have a default "system" version, then I install `nvm` and use it when I need distinct versions.

Comment (3cp) [inline .install-osx.sh:9]: that means travis linux box has node installed by default?

Comment (3cp) [inline .install-osx.sh:9]: yes, it does. https://docs.travis-ci.com/user/ci-environment/#Runtimes

**Comment by 3cp** [comment]
Ideally, it's better to test .install-linux.sh on osx too. But I don't know how to create 2 variants on osx while keep 1 variant on linux in .travis.yml.

**Comment by derekstavis** [comment]
Would you mind squashing fixup commits? I m willing to merge this so I can work on improving testing and load speed on OS X.

**Comment by 3cp** [comment]
done.
